pkg/diagnostics/cluster/aggregated_logging/services.go

```diff
@@ -0,0 +1,49 @@
+package aggregated_logging
+
+import (
+	"fmt"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+var loggingServices = sets.NewString("logging-es", "logging-es-cluster", "logging-es-ops", "logging-es-ops-cluster", "logging-kibana", "logging-kibana-ops")
+
+const serviceNotFound = `
+Expected to find '%s' among the logging services for the project but did not.  Depending 
+upon the options chosen during installation, these services (e.g. logging-es-ops) may not have been provided.
+`
+
+// checkServices looks to see if the aggregated logging services exist
+func checkServices(r diagnosticReporter, adapter servicesAdapter, project string) {
+	r.Info("AGL0200", fmt.Sprintf("Checking for services in project '%s' with selector '%s'", project, loggingSelector.AsSelector()))
+	serviceList, err := adapter.services(project, kapi.ListOptions{LabelSelector: loggingSelector.AsSelector()})
+	if err != nil {
+		r.Error("AGL0205", err, fmt.Sprintf("There was an error while trying to retrieve the logging services: %s", err))
+		return
+	}
+	foundServices := sets.NewString()
+	for _, service := range serviceList.Items {
+		foundServices.Insert(service.ObjectMeta.Name)
+		r.Debug("AGL0210", fmt.Sprintf("Retrieved service '%s'", service.ObjectMeta.Name))
+	}
+	for _, service := range loggingServices.List() {
+		if foundServices.Has(service) {
+			checkServiceEndpoints(r, adapter, project, service)
+		} else {
+			r.Warn("AGL0215", nil, fmt.Sprintf(serviceNotFound, service))
+		}
+	}
+}
+
+// checkServiceEndpoints validates if there is an available endpoint for the service.
+func checkServiceEndpoints(r diagnosticReporter, adapter servicesAdapter, project string, service string) {
+	endpoints, err := adapter.endpointsForService(project, service)
+	if err != nil {
+		r.Warn("AGL0220", err, fmt.Sprintf("Unable to retrieve endpoints for service '%s': %s", service, err))
+		return
+	}
+	if len(endpoints.Subsets) == 0 {
+		r.Warn("AGL0225", nil, fmt.Sprintf("There are no endpoints found for service '%s'. This may not be a problem if the backing pods were not deployed (e.g. ops).", service))
```

Comment:
Also, a little context... what does this mean to the end user? Need to let them know it means there are no pods serviced by this service, so the component is not functioning.